src/lib/analysers/meta/main.py

```diff
@@ -0,0 +1,71 @@
+from lib.common.analyser import Analyser
+from lib.common.analyser import paths_to_components
+from lib.common.get_module import get_module
+import os
+from shutil import copyfile, rmtree
+
+
+class MetaAnalyser(Analyser):
+    def setup_run(self, config):
+
+        self.child_analysers = []
+        whitelist = self.CONFIG["elements_in"]
+
+        cmps = paths_to_components(whitelist)
+        self.selector = cmps[0][0]
+
+        child_modules = config["modules"]
+        for module in child_modules:
+            child_name = module["name"]
+            child_config = module["config"]
+            ChildAnalyser = get_module("analyser", child_name)
+            if ChildAnalyser is None:
+                raise Exception(
+                    f"The module you have specified, {child_name}, does not exist"
+                )
+            child_analyser = ChildAnalyser(child_config, child_name, self.FOLDER)
+            child_analyser.setup_run(child_config)
+            self.child_analysers.append(child_analyser)
+            self.logger(f"Setup child analyser {child_name}")
```

Comment:
One thing to note here is that, in the way it's currently coded, chaining analysers together via `meta` actually slightly modifies the overall life cycle. All `setup_run`s for analysers in the chain are run up front, and then `analyse_element`s are called in sequence appropriately.

As we don't have hard guarantees on what can and can't happen in `setup_run` yet, this could actually get weird if the logic in `setup_run` relies on a particular state of the mtriage file system, e.g. elements that are about to be analysed.

I think what this actually flags is that *we want to make it clear that `setup_run` shouldn't reference the mtriage file system* in any way, as indicated by its arguments. I don't think we can isolate it to Python runtime logic entirely, i.e. only setting up constants... as in the `pytorch` analyser we want to call setup functions that, for all I know, contain a load of side effects.